src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
+   * @inner
+   */
+  async validateAndRenewAuthToken() {
+    const now = new Date();
+    const timestampNow = Math.floor(now.getTime() / 1000);
+
+    const validateJWTExpireDate = (token) => {
+      const base64Url = token.split('.')[1];
+      const base64 = base64Url.replace('-', '+').replace('_', '/');
+      const decodedData = JSON.parse(Buffer.from(base64, 'base64').toString('binary'));
+
+      // If the token will expire in the next 60 seconds (or has already expired)
+      const delta = 60;
+      if (timestampNow + delta > decodedData.exp) {
+        return false;
+      }
+
+      return true;
+    }
+
+    if (!this.authToken || !validateJWTExpireDate(this.authToken)) {
+      const sign = this.getAuthSign(timestampNow);
+      const response = await walletApi.createAuthToken(timestampNow, this.xpub!, sign);
+      if (response.status === 200 && response.data.success === true) {
+        this.authToken = response.data.token;
+      }
+    }
+  }
+
   /**
    * Send a transaction from an array of outputs and inputs
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async sendManyOutputsTransaction(outputs, options = { inputs: [], changeAddress: null }) {
-    return await this.sendTxProposal(outputs, options);
+  async sendManyOutputsTransaction(outputs: OutputRequestObj[], options: { inputs?: InputRequestObj[], changeAddress?: string } = {}): Promise<TxProposalUpdateResponseData> {
+    this.checkWalletReady();
+    const newOptions = Object.assign({
+      inputs: [],
+      changeAddress: null
+    }, options);
+    return await this.createAndSendTx(outputs, newOptions);
   }
 
   /**
    * Send a transaction to a single output
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async sendTransaction(address, value, options: { token: string | null, changeAddress: string | null } = { token: '00', changeAddress: null }) {
+  async sendTransaction(address: string, value: number, options: { token?: string, changeAddress?: string } = {}): Promise<TxProposalUpdateResponseData> {
+    this.checkWalletReady();
     const newOptions = Object.assign({
       token: '00',
       changeAddress: null
     }, options);
     const { token, changeAddress } = newOptions;
     const outputs = [{ address, value, token }];
-    return await this.sendTxProposal(outputs, { inputs: [], changeAddress });
+    return await this.createAndSendTx(outputs, { inputs: [], changeAddress });
   }
 
   /**
-   * Send a transaction proposal
+   * Send transaction
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async sendTxProposal(outputs, options = { inputs: [], changeAddress: null }) {
-    const newOptions = Object.assign({
+  async createAndSendTx(outputs: OutputRequestObj[], options: { inputs?: InputRequestObj[], changeAddress?: string } = {}): Promise<TxProposalUpdateResponseData> {
```

Comment:
Giant method! Should we split some parts in separate methods for clarity and simpler unittests?

Reply:
We had a great discussion in terms of interfaces of the new facades and I've done a bigger refactor in the sendTransaction code in the new branch (the one that matches the signatures). I prefer keeping like this here and just do the refactor there, what do you think?

Reply:
As it has already been done there, no need to do it here. Thanks!